### PR TITLE
feat: Use `klona` library for deep clone

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/clone.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
 /**
  * Creates a deep clone of an object.
  * @deprecated Use {@link structuredClone} instead.
  * @param input The object to clone.
  */
-export function clone<T>(input: T | null = null): T {
-  const stringifiedInput = quickStringify(input);
-  return stringifiedInput ? JSON.parse(stringifiedInput) : null;
+export function clone<T = unknown>(input: T): T {
+  return klona(input);
```
Comment (viceice) [inline lib/util/clone.ts:9]: what's the benefit to use klona? if I remember correctly it has problems with circular references.

Comment (zharinov) [reply lib/util/clone.ts:9]: Yes, but I'm not sure we have them, nor we need to have it by design. So, let's just treat them as bugs when `RangeError` has encountered.

Comment (zharinov) [reply lib/util/clone.ts:9]: In exchange, we'll receive speed for this particular operation

Comment (viceice) [reply lib/util/clone.ts:9]: https://github.com/lukeed/klona/issues/40

Comment (viceice) [reply lib/util/clone.ts:9]: https://github.com/lukeed/klona#klonalite

we probably need the lite variant to support date and regex

Comment (zharinov) [reply lib/util/clone.ts]: Fixed

**Review by rarkins — CHANGES_REQUESTED** [review]
Can you add a try/catch and fall back to old approach with a warning if klona throws?

File: lib/util/clone.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
  * @deprecated Use {@link structuredClone} instead.
  * @param input The object to clone.
  */
-export function clone<T>(input: T | null = null): T {
-  const stringifiedInput = quickStringify(input);
-  return stringifiedInput ? JSON.parse(stringifiedInput) : null;
+export function clone<T = unknown>(input: T): T {
+  try {
+    return klona(input);
+  } catch (error) {
```
Comment (rarkins) [inline lib/util/clone.ts:12]:
```suggestion
  } catch (err) {
    logger.warn({ err }, 'error cloning object');
```

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

File: lib/modules/datasource/npm/__snapshots__/index.spec.ts.snap [hunk 3]
```diff
@@ -28,10 +34,16 @@ exports[`modules/datasource/npm/index should fetch package info from npm 1`] = `
   "registryUrl": "https://registry.npmjs.org",
```
Comment (Hirammm80) [inline lib/modules/datasource/npm/__snapshots__/index.spec.ts.snap:34]: c24959507180ed88cd8ebccdad52ea2a445bfc9d

**Review by Hirammm80 — APPROVED** [review]

**Pull request merged**